src/app/tabs/media.cpp

```diff
@@ -223,3 +232,206 @@ QWidget *RadioPlayerSubTab::controls_widget()
+        if (player->playlist()->currentIndex() < 0) player->playlist()->setCurrentIndex(0);
+        player->playlist()->previous();
+        player->play();
+    });
+    theme->add_button_icon("skip_previous", previous_button);
+    layout->addStretch();
+    layout->addWidget(previous_button);
+
+    QPushButton *play_button = new QPushButton(widget);
+    play_button->setFlat(true);
+    play_button->setCheckable(true);
+    play_button->setChecked(false);
+    play_button->setIconSize(Theme::icon_84);
+    connect(play_button, &QPushButton::clicked, [player = this->player, play_button](bool checked = false) {
+        play_button->setChecked(!checked);
+        if (checked)
+            player->play();
+        else
+            player->pause();
+    });
+    connect(this->player, &QMediaPlayer::stateChanged,
+            [play_button](QMediaPlayer::State state) { play_button->setChecked(state == QMediaPlayer::PlayingState); });
+    theme->add_button_icon("pause", play_button, "play");
+    layout->addStretch();
+    layout->addWidget(play_button);
+
+    QPushButton *forward_button = new QPushButton(widget);
+    forward_button->setFlat(true);
+    forward_button->setIconSize(Theme::icon_84);
+    connect(forward_button, &QPushButton::clicked, [player = this->player]() {
+        player->playlist()->next();
+        player->play();
+    });
+    theme->add_button_icon("skip_next", forward_button);
+    layout->addStretch();
+    layout->addWidget(forward_button);
+    layout->addStretch();
+
+    return widget;
+}
+
+QString LocalPlayerSubTab::durationFmt(int total_ms)
+{
+    int mins = (total_ms / (1000 * 60)) % 60;
+    int secs = (total_ms / 1000) % 60;
+
+    return QString("%1:%2").arg(mins, 2, 10, QChar('0')).arg(secs, 2, 10, QChar('0'));
+}
+
+void LocalPlayerSubTab::populate_dirs(QString path, QListWidget *dirs_widget)
+{
+    dirs_widget->clear();
+    QDir current_dir(path);
+    QFileInfoList dirs = current_dir.entryInfoList(QDir::AllDirs | QDir::Readable);
+    for (QFileInfo dir : dirs) {
+        if (dir.fileName() == ".") continue;
+
+        QListWidgetItem *item = new QListWidgetItem(dir.fileName(), dirs_widget);
+        if (dir.fileName() == "..") {
+            item->setText("↲");
+
+            if (current_dir.isRoot()) item->setFlags(Qt::NoItemFlags);
+        }
+        else {
+            item->setText(dir.fileName());
+        }
+        item->setFont(Theme::font_16);
+        item->setData(Qt::UserRole, QVariant(dir.absoluteFilePath()));
+    }
+}
+
+void LocalPlayerSubTab::populate_tracks(QString path, QListWidget *tracks_widget)
+{
+    QStringList tracks = QDir(path).entryList(QStringList() << "*.mp3", QDir::Files | QDir::Readable);
+    for (QString track : tracks) {
+        if (this->player->playlist()->addMedia(QMediaContent(QUrl::fromLocalFile(path + '/' + track)))) {
+            TagLib::FileRef f(std::string(path.toStdString() + "/" + track.toStdString()).c_str());
```

Comment:
was testing out some things with getting the mp3 metadata here, but this will be fleshed out in a later updated